### PR TITLE
fix(install): preflight Node.js and npm prerequisites in install.ps1

### PR DESCRIPTION
**Pull request by wzc1753**
## What Changed
- Added native preflight environment checks in `install.ps1`:
  - Validates Node.js presence in PATH (`Get-Command node -ErrorAction SilentlyContinue`).
  - Validates Node.js major version is >= 18 (`node -v` parsing) aligned with `package.json` `"engines"`.
  - Validates `npm` presence when installing from a git clone lacking `node_modules`.
  - Uses `[Console]::Error.WriteLine` + explicit `exit` codes to preserve process exit codes and prevent noisy PowerShell `ActionPreferenceStopException` stack traces under `$ErrorActionPreference = 'Stop'`.
- Added unit and regression tests in `tests/scripts/install-ps1.test.js`:
  - Test verifying clean rejection and actionable error message when Node.js is missing from PATH.
  - Test verifying clean rejection when Node.js version is older than 18 (mocked `v16.20.0`).
  - Hardened PowerShell binary path resolution with `fs.realpathSync.native` and isolated PATH fixtures across platforms.

## Why This Change
On Windows systems, running `install.ps1` without Node.js or with an unsupported Node version (< 18) previously resulted in raw PowerShell terminating exceptions (e.g. `npm: The term 'npm' is not recognized` or syntax errors inside modern JavaScript runtime) without clear guidance for the user. These preflight checks fail fast with actionable instructions and official download URLs before any target directory or configuration is mutated.

## Testing Done
- [x] Manual testing completed on Windows with pwsh 7 and Windows PowerShell 5.1
- [x] Automated tests pass locally (`node tests/scripts/install-ps1.test.js` - 6 passed, 0 failed; `node tests/scripts/install-apply.test.js` - 42 passed, 0 failed; `node tests/scripts/install-plan.test.js` - 11 passed, 0 failed)
- [x] Edge cases considered and tested (isolated PATH, non-standard version strings, StrictMode scalar handling)

## Type of Change
- [x] `fix:` Bug fix
- [ ] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [x] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [x] Shell scripts pass shellcheck (if applicable)
- [x] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [ ] Updated relevant documentation
- [x] Added comments for complex logic
- [ ] README updated (if needed)

### CODE REVIEW

File: install.ps1 [hunk 1]
```diff
@@ -34,15 +34,31 @@ while ($true) {
...
+if (-not (Get-Command node -ErrorAction SilentlyContinue)) {
+    [Console]::Error.WriteLine('[ECC] Node.js is required but was not found in PATH. Please install Node.js 18 or newer: https://nodejs.org')
+    exit 1
+}
+
+$nodeVersion = (& node -v 2>$null | Select-Object -First 1)
+if ($nodeVersion -match '^\s*v?(\d+)\.' -and [int]$Matches[1] -lt 18) {
+    [Console]::Error.WriteLine("[ECC] Node.js 18 or newer is required (found $nodeVersion). Please update Node.js: https://nodejs.org")
+    exit 1
+}
```
[the remark this answers is not in the document]
Comment (wzc1753) [reply install.ps1:53]: Addressed in 5c78d38. The script now fails closed with a clear error message when Node.js version cannot be determined or parsed.

File: tests/scripts/install-ps1.test.js [hunk 2]
```diff
@@ -63,6 +74,19 @@ function run(powerShellCommand, args = [], options = {}) {
     USERPROFILE: options.homeDir || process.env.USERPROFILE,
   };
 
+  if (options.env) {
+    for (const [key, val] of Object.entries(options.env)) {
+      if (key.toLowerCase() === 'path') {
+        for (const k of Object.keys(env)) {
+          if (k.toLowerCase() === 'path') {
+            delete env[k];
+          }
```
[the remark this answers is not in the document]
Comment (wzc1753) [reply tests/scripts/install-ps1.test.js]: Addressed in 5c78d38. The child environment is now constructed immutably using Object.fromEntries without mutating the base environment in place.

File: install.ps1 [hunk 3]
```diff
@@ -34,15 +34,31 @@ while ($true) {
...
 $installerScript = Join-Path -Path (Join-Path -Path $scriptDir -ChildPath 'scripts') -ChildPath 'install-apply.js'
 
+# ponytail: preflight Node.js presence and version (>= 18)
+if (-not (Get-Command node -ErrorAction SilentlyContinue)) {
+    [Console]::Error.WriteLine('[ECC] Node.js is required but was not found in PATH. Please install Node.js 18 or newer: https://nodejs.org')
+    exit 1
+}
+
+$nodeVersion = (& node -v 2>$null | Select-Object -First 1)
+if ($nodeVersion -match '^\s*v?(\d+)\.' -and [int]$Matches[1] -lt 18) {
```
[the remark this answers is not in the document]
Comment (wzc1753) [reply install.ps1]: Addressed in 5c78d38. The script now fails closed when the Node.js version output is empty or unparseable, and added an automated regression test covering unparseable version strings.

File: tests/scripts/install-ps1.test.js [hunk 4]
```diff
@@ -160,6 +181,109 @@ function runTests() {
...
+    }
+  })) passed++; else failed++;
+
+  if (!powerShellCommand) {
+    console.log('  - skipped unparseable Node.js preflight test; PowerShell is not available in PATH');
+  } else if (test('rejects execution when Node.js version is unparseable', () => {
+    const mockBinDir = createTempDir('install-ps1-invalid-node-');
+    try {
+      if (process.platform === 'win32') {
+        fs.writeFileSync(path.join(mockBinDir, 'node.cmd'), '@echo unexpected-node-output\r\n');
```
[the remark this answers is not in the document]
Comment (wzc1753) [reply tests/scripts/install-ps1.test.js]: Addressed in commit f6eeb46: Extracted \INVALID_NODE_VERSION_OUTPUT\ constant to comply with coding guidelines and avoid repeated literals.

File: install.ps1 [hunk 5]
```diff
@@ -34,15 +34,37 @@ while ($true) {
...
+    exit 1
+}
+
 # Auto-install Node dependencies when running from a git clone
 $nodeModules = Join-Path -Path $scriptDir -ChildPath 'node_modules'
 if (-not (Test-Path -LiteralPath $nodeModules)) {
+    if (-not (Get-Command npm -ErrorAction SilentlyContinue)) {
+        [Console]::Error.WriteLine('[ECC] npm is required to install dependencies but was not found in PATH.')
+        exit 1
+    }
```
[the remark this answers is not in the document]
Comment (wzc1753) [reply install.ps1:61]: Addressed in commit f6eeb46: Added regression test cases in \	ests/scripts/install-ps1.test.js\ covering missing npm preflight rejection as well as nonzero \
pm install\ error messaging and exit-code propagation.

**Comment by chenhz01** [comment]
Two things worth fixing before merge, both visible in the diff. First, the preflight block carries a stray marker comment — `# ponytail: preflight Node.js presence and version (>= 18)` — which reads like leftover scaffolding and should just be `# Preflight: ...`. Second, `$minimumNodeMajor = 18` hardcodes a number that `package.json` `engines` already owns; when the floor moves to 20 the two will disagree and install.ps1 will gate on the wrong version. Either parse the floor at preflight time (`node -p "require('./package.json').engines.node"`) or add a loud cross-reference comment next to both. The `[Console]::Error.WriteLine` + explicit `exit` pattern is the right call — it sidesteps the `ActionPreferenceStopException` noise this PR describes — and the missing-node and v16 rejection tests cover the main paths. Happy to take the comment and version-source cleanup if useful.
